internal/test/builder/builders.go

```diff
@@ -1166,7 +1427,7 @@ func (m *MachineDeploymentBuilder) WithBootstrapTemplate(ref *unstructured.Unstr
 	return m
 }
 
-// WithInfrastructureTemplate adds the passed unstructured object to the MachineDeployment builder as an infrastructureMachineTemplate.
+// WithInfrastructureTemplate adds the passed unstructured object to the MachineDeployment builder as an infrastructureMachinePoolTemplate.
```

Comment:
This is for MachineSets :-)

```suggestion
// WithInfrastructureTemplate adds the passed unstructured object to the MachineDeployment builder as an infrastructureMachineTemplate.
```